### ISSUE TITLE
Private content

### ISSUE THREAD

**Issue by JoshHeng** [WarwickTechCrew/website#86]
We have a need of a way to show private information on the wiki that is only accessible to either (a) Tech Crew members or (b) Uni of Warwick students/staff. This could be info that we may not be at liberty to disclose to the public such as price lists, third party info, etc.

My proposed solution is:
* We create a new private repository (only accessible to the wiki GitHub team) called website-private
* This repository handles authentication of users and provides an API which, once provided with a valid JWT token, returns JSON or markdown for the requested resource
* Authentication is done via Warwick SSO or via shared passwords given by exec. There are two levels of authorisation - UoW staff/students, and TC members (checked via the SU membership API)
* We add support for private resources in frontmatter, and a private markdown block. From the wiki side, these will just accept a path to the resource on the private API
* When a user loads the page, a client-side script will check if the user is authenticated. If they are, it will request and display the relevant info. If not, it will show a 'Login Required' space with a link to login with Warwick SSO

Any thoughts on this?

**Comment by k-shar** [comment]
is doing AdamRMS CMS pages any more distributed than a new wiki?
I feel its more combined if we have wiki for public and adamrms for private, given adamrms already contains the role application system and our information about quotes + equipment ect

**Comment by JoshHeng** [comment]
> is doing AdamRMS CMS pages any more distributed than a new wiki? I feel its more combined if we have wiki for public and adamrms for private, given adamrms already contains the role application system and our information about quotes + equipment ect

So it would still be the same wiki, except restricted content would just show a login box until you are logged in. The advantage of this over AdamRMS is that headings etc. would still show up in the wiki search (providing the headings are in the un-redacted area), and shortlinks etc. would also still work

**Comment by k-shar** [comment]
what kind of content would acc go as restricted? i cant think of any long-form wiki content that would need to be restricted? Im just thinking adamrms to host internal documents

**Comment by JoshHeng** [comment]
* Price lists
* (potentially) some equipment lists
* Discord invites
* AdamRMS business codes
* Internal uni procedures (e.g. floorplans, permits, etc.) that probably shouldn't be completely public
* Internal WAC procedures and internal info (e.g. internal doors/BoH map, marketing guidelines, etc.)

**Comment by k-shar** [comment]
Yeah so like the first 3 i feel are better as one liner links that can be in a CMS
buidness code should start going in email probably

i guess overall im just thinking is it worth doing all the dev to make private pages given we already have a members only area. questions about maintainability in the future ect. Is it not better to expand on the existing adamrms system

**Comment by JoshHeng** [comment]
I'm not a huge fan of spreading long term info across two platforms as it can be unclear where to look for things, especially if we're pushing search as the main way to find information.

I also don't really like how the AdamRMS CMS works, would prefer still having everything in markdown if possible so people only need to learn and remember to update one thing. @adamskrz suggested having a repo and then system to automatically push to AdamRMS through the API, but I think that would be an equivalently complex (and perhaps more cursed) method.

First three would be restricted links in the 'resources' section, the others would be restricted blocks in the wiki.

**Comment by JoshHeng** [comment]
The links are less of an issue though, it's just there's been several things that I've thought could go on the wiki that we can't do because we don't want it completely public. I think @adamskrz might have had the same thoughts at some point

**Comment by k-shar** [comment]
Hmm okay, 
- whats the hosting, UWCS? is this a bit sketchy future proofing
- whats the difference between staff/student auth and member auth, is it a bit weird for tc wiki to turn into uni of warwick wiki?
- oh and yeah what are the other members only things ?

**Comment by JoshHeng** [comment]
1. Hosting probably on Vercel, so should be fairly future proof. We can do it on the TC GitHub user (I think we have one? Otherwise can make one on the shared gmail)
2. Staff/student auth might be things we want to open to all Warwick Drama such as TC price list and some WAC things. Member auth would be things like finance, internal uni procedures and resources (e.g. floorplans)
3. It would be possible to extend it yes, not sure what currently. A cool plan down the line could be stuff like scanning ur uni card to 'sign in' or out of get-ins/outs, but this is not that related.

**Comment by JoshHeng** [comment]
Oh also need to add:
- DCRS radio pricing
- SU tech services pricing 
- WAC logos & brand guide

**Comment by adamskrz** [comment]
I have a coursework to do so finally getting around to wiki stuff!

> @adamskrz suggested having a repo and then system to automatically push to AdamRMS through the API, but I think that would be an equivalently complex (and perhaps more cursed) method.

The AdamRMS CMS API just uses HTML as far as I can tell. So it would just be a script to render Markdown to HTML, and push it to a CMS page (or make a new one if it doesn't exist). I agree with Kishan about making a whole new auth system and infra for the couple of price lists and contacts being really unnecessary.
Thinking about it now - we could just let everyone have edit perms to CMS pages on AdamRMS and skip git altogether. Though I guess some hypothetical bad actor could delete it all? I don't think theres any edit history there.

There is something pretty compelling about the idea of there just being a block to load hidden content in the main wiki though.

> I'm not a huge fan of spreading long term info across two platforms as it can be unclear where to look for things, especially if we're pushing search as the main way to find information.

I'm not a fan of having more web platforms and systems to maintain :P (especially ones that rely on third parties like the uni). Is there actually any reason to use accounts or whatever instead of just one shared password? The main thing here is just to prevent this info being discoverable by randoms on the internet right.
In terms of discoverability, there's nothing stopping us linking to an AdamRMS CMS page from the public wiki anyway, keeping everything easy to find there?

I was reading https://dev.to/charca/password-protection-for-cloudflare-pages-8ma and this approach with Cloudflare Workers seems pretty simple, which makes me think maybe your idea could be not bad after all.

**Comment by JoshHeng** [comment]
![Image](https://github.com/user-attachments/assets/6992a688-834d-401d-aff3-2e3755197c54)

I'm probably going to go ahead with implementing private content support using this architecture.
